packages/Babylonian-Printbugger.package/BPSwimlane.class/instance/toggleAnnotationVisibilityButtonClicked..st

```diff
@@ -0,0 +1,11 @@
+private - ui
+toggleAnnotationVisibilityButtonClicked: aTraceValue
+
+	(self isTraceValueVisible: aTraceValue)
+		ifTrue: [context ignoreAnnotation: aTraceValue belongingAnnotationId for: example.]
+		ifFalse: [context unignoreAnnotation: aTraceValue belongingAnnotationId for: example].
+
+	annotationIdToFlash := aTraceValue belongingAnnotationId.
+	self submorphsDo: [:aSubmorph | aSubmorph delete].
+	self visualize. 
+	annotationIdToFlash := nil.
```

Comment:
could you explain this part to me? why to you set the annotationidToFlash, just to delete all morphs of the swimlane and then setting it to nil?

Reply:
`BPLabelMorph` is really just the label with the hide/show and delete button above a displayed value in each line. The label only knows the traceValue it belongs to, but not the other tracevalues for the same annotation. That is why the visibility-toggle button is created in the Swimlane and then passed to the `BPLabelMorph` to be able to call this method when clicked and have access to `visualize` and the swimlane's submorphs.

So, since we want to flash the labels that belong to the same annotation that gets toggled, we need to know in the `visualize` method what traceValues we actually need to flash (since as I said, a single traceValue has no connection to its other traceValues of the same annotation for this - at least as far as I know).

So, the process here is:
Store the annotation that's about to flash, so the traceValues know which of them need to flash when visualizing. Afterwards the variable is set to `null`, so those values don't flash on the next rerendering that was triggered by something else then a click on the visibility-toggle button.

Does that make sense?

Reply:
yes. move the `self submorphsDo: [:aSubMorph | aSubmorph abandon]` to visualize though.  (abandon, not delete)